### PR TITLE
Add helm lint action

### PR DESCRIPTION
**Pull request by MindTooth**
## Description

Second attempt with newer version.  Supersedes https://github.com/pglombardo/PasswordPusher/pull/1569.

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've written tests (if applicable) for all new methods and classes that I created. (`rake test`)
- [ ] I've added documentation as necessary so users can easily use and understand this feature/fix.

### CODE REVIEW

**Comment by MindTooth** [comment]
Seeing as it does not work.  I posted a comment to an issue someone else has reported too. https://github.com/helm/chart-testing-action/issues/108

File: .github/workflows/lint-helm.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+name: Lint and Test Charts
+
+on: pull_request
+
+jobs:
+  lint-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (cpanato) [inline .github/workflows/lint-helm.yaml:11]: bump to v4

File: .github/workflows/lint-helm.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          python-version: "3.10"
+          check-latest: true
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.6.0
+
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
```
Comment (cpanato) [inline .github/workflows/lint-helm.yaml:31]:
```suggestion
          changed=$(ct list-changed --config ./ct.yaml --target-branch ${{ github.event.repository.default_branch }})
```

File: .github/workflows/lint-helm.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          if [[ -n "$changed" ]]; then
+            echo "changed=true" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Run chart-testing (lint)
+        if: steps.list-changed.outputs.changed == 'true'
+        run: ct lint --target-branch ${{ github.event.repository.default_branch }}
```
Comment (cpanato) [inline .github/workflows/lint-helm.yaml:38]:
```suggestion
        run: ct lint --config ./ct.yaml --target-branch ${{ github.event.repository.default_branch }}
```

can you add this and try again?

File: .github/workflows/lint-helm.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Create kind cluster
+        if: steps.list-changed.outputs.changed == 'true'
+        uses: helm/kind-action@v1.8.0
+
+      - name: Run chart-testing (install)
+        if: steps.list-changed.outputs.changed == 'true'
+        run: ct install --target-branch ${{ github.event.repository.default_branch }}
```
Comment (cpanato) [inline .github/workflows/lint-helm.yaml:46]:
```suggestion
        run: ct install --config ./ct.yaml --target-branch ${{ github.event.repository.default_branch }}
```

**Comment by MindTooth** [comment]
@pglombardo is this something that you would accept now that it works?

**Comment by pglombardo** [comment]
> @pglombardo is this something that you would accept now that it works?

Absolutely and much appreciated!  Give me a bit to get some time and go through the PR (and learn the Helm basics).

**Comment by pglombardo** [comment]
Looks good to me @MindTooth.  Should we (or is there any value) in publishing these Helm charts to ArtifactHub?

**Comment by MindTooth** [comment]
Maybe host ourself using https://github.com/helm/chart-releaser?  Can then forgo any need for auth against ArtifactHub. 🤷🏻‍♂️

**Comment by MindTooth** [comment]
We should also update it with all parameters as of late.  Can do that when this is merged.  Start to treat helm as a valid option.

**Pull request merged**